File: src/main/java/client/inventory/slots/ItemSlotBundle.java
package client.inventory.slots;

import java.util.Objects;

public class ItemSlotBundle extends ItemSlot {

    private short number, maxNumber, attribute;
    private String title = "";

    public short getNumber() {
        return number;
    }

    public void setNumber(short number) {
        this.number = number;
    }

    public short getMaxNumber() {
        return maxNumber;
    }

    public void setMaxNumber(short maxNumber) {
        this.maxNumber = maxNumber;
    }

    public short getAttribute() {
        return attribute;
    }

    public void setAttribute(short attribute) {
        this.attribute = attribute;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ItemSlotBundle that = (ItemSlotBundle) o;
        return templateId == that.templateId &&
                maxNumber == that.maxNumber &&
                attribute == that.attribute &&
                title.equals(that.title) &&
                cashItemSN == that.cashItemSN &&
                expire == that.expire;
    }

    @Override
    public int hashCode() {
        return Objects.hash(templateId, maxNumber, attribute, title, cashItemSN, expire);
    }

    @Override
    public String toString() {
        return super.toString() + "\nItemSlotBundle{" +
                "number=" + number +
                ", maxNumber=" + maxNumber +
                ", attribute=" + attribute +
                ", title='" + title + '\'' +
                '}';
    }
}

File: src/main/java/client/inventory/slots/ItemSlotBundle.kt
package client.inventory.slots

import java.util.*

class ItemSlotBundle : ItemSlot() {

    var number: Short = 0
    var maxNumber: Short = 0
    var attribute: Short = 0
    var title = ""

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as ItemSlotBundle
        return templateId == that.templateId && maxNumber == that.maxNumber && attribute == that.attribute && title == that.title && cashItemSN == that.cashItemSN && expire == that.expire
    }

    override fun hashCode(): Int {
        return Objects.hash(templateId, maxNumber, attribute, title, cashItemSN, expire)
    }

    override fun toString(): String {
        return "${super.toString()}\nItemSlotBundle{number=$number, maxNumber=$maxNumber, attribute=$attribute, title='$title'}"
    }
}